### DISCUSSION TITLE
detectBrowserLanguage and alwaysRedirect unclear routing logic

### DISCUSSION THREAD

**Discussion by spiritius** [nuxt-modules/i18n#3153] [General]
### Environment

```
- Operating System: Darwin
- Node Version:     v22.3.0
- Nuxt Version:     3.12.4
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  npm@10.8.1
- Builder:          -
- User Config:      compatibilityDate, runtimeConfig, srcDir, dir, app, css, modules, i18n, vite
- Runtime Modules:  @nuxtjs/tailwindcss@6.12.1, @nuxtjs/i18n@8.3.3
- Build Modules:    -
```

### Reproduction

- code: https://stackblitz.com/edit/bobbiegoede-nuxt-i18n-starter-4bwqjd?file=nuxt.config.ts
- preview: https://bobbiegoedenuxti18nstarter4bwq-korv--3000--41692973.local-credentialless.webcontainer.io/

### Describe the bug

Hi!

When using the `prefix_and_default` strategy and the `detectBrowserLanguage` option, the redirect logic is unclear.

**Current behavior**:

`site.com/` → click `NL` → `site.com/nl` → change url manually to `site.com/` → get `site.com/nl`

**What I expected** from the parameter `alwaysRedirect: false`:

`site.com/` → click `NL` → `site.com/nl` → change url manually to `site.com/` → get `site.com/`

Am I doing something wrong? Is there any way to fix this in nuxt.config.ts?

### Additional context

```jsx
// nuxt.config.ts
i18n: {
  vueI18n: '~/app/i18n/i18n.config.ts',
  baseUrl: process.env.BASE_URL,
  locales: [
    { code: 'en', iso: 'en-US', name: 'English' },
    { code: 'nl', iso: 'nl-NL', name: 'Nederlands' }
  ],
  defaultLocale: 'en',
  strategy: 'prefix_and_default',
  detectBrowserLanguage: {
    fallbackLocale: 'en',
    useCookie: true,
    cookieKey: 'lang',
    alwaysRedirect: false,
    redirectOn: 'all'
  }
}

// i18n.config.ts
import en from './locales/en.json'
import nl from './locales/nl.json'

export default defineI18nConfig(() => ({
  legacy: false,
  locale: 'en',
  fallbackLocale: 'en',
  globalInjection: true,
  messages: {
    en,
    nl
  },
  missingWarn: false, 
  fallbackWarn: false,
}))
```

**Comment by BobbieGoede** [comment]
Sorry about that, I think this seems to be an issue after all.. Creating a new issue 😅
